test/filter_tests/src/filter/test_filter_biquad_sat_s32.c

```diff
@@ -0,0 +1,217 @@
+        for(unsigned int j = 0; j < filter.biquad_count; j++)
+            filter.coef[i][j] = 0x40000000;
+        for(int j = filter.biquad_count; j < 8; j++)
+            filter.coef[i][j] = 0;
+    }
+
+    int32_t res = filter_biquad_sat_s32(&filter, 100);
+    TEST_ASSERT_EQUAL(100, res);
+
+    res = filter_biquad_sat_s32(&filter, 50);
+    TEST_ASSERT_EQUAL(250, res);
+
+    res = filter_biquad_sat_s32(&filter, 4000);
+    TEST_ASSERT_EQUAL(4500, res);
+}
+
+TEST(filter_biquad_sat_s32, case1)
+{
+    filter_biquad_s32_t filter;
+
+    filter.biquad_count = 2;
+
+    memset(filter.state, 0, sizeof(filter.state));
+
+    for(int i = 0; i < 5; i++){
+        for(unsigned int j = 0; j < filter.biquad_count; j++)
+            filter.coef[i][j] = 0x40000000;
+        for(int j = filter.biquad_count; j < 8; j++)
+            filter.coef[i][j] = 0;
+    }
+
+    // x[n  ] = 100
+    // y[0][n] = 1.0 * (  100 + 0 + 0 + 0 + 0  ) = 100
+    // y[1][n] = 1.0 * (  100 + 0 + 0 + 0 + 0  ) = 100
+    int32_t res = filter_biquad_sat_s32(&filter, 100);
+    TEST_ASSERT_EQUAL(100, res);
+
+    // x[n  ] = 50
+    // y[0][n] = 1.0 * (  50 + 100 +   0 + 100 +   0  ) = 250
+    // y[1][n] = 1.0 * ( 250 + 100 +   0 + 100 +   0  ) = 450
+    res = filter_biquad_sat_s32(&filter, 50);
+    TEST_ASSERT_EQUAL(450, res);
+
+    // x[n  ] = 4000
+    // y[0][n] = 1.0 * ( 4000 +   50 +  100 +  250 +  100  ) = 4500
+    // y[1][n] = 1.0 * ( 4500 +  250 +  100 +  450 +  100  ) = 5400
+    res = filter_biquad_sat_s32(&filter, 4000);
+    TEST_ASSERT_EQUAL(5400, res);
+}
+
+TEST(filter_biquad_sat_s32, case2)
+{
+    filter_biquad_s32_t filter;
+
+    filter.biquad_count = 3;
+
+    memset(filter.state, 0, sizeof(filter.state));
+
+    for(int i = 0; i < 5; i++){
+        for(unsigned int j = 0; j < filter.biquad_count; j++)
+            filter.coef[i][j] = (j==(filter.biquad_count-1))? 0x20000000 : 0x40000000;
+        for(int j = filter.biquad_count; j < 8; j++)
+            filter.coef[i][j] = 0;
+    }
+
+    // x[n] = 100
+    // y[0][n] = 1.0 * (  100 + 0 + 0 + 0 + 0  ) = 100
+    // y[1][n] = 1.0 * (  100 + 0 + 0 + 0 + 0  ) = 100
+    // y[2][n] = 0.5 * (  100 + 0 + 0 + 0 + 0  ) =  50
+    int32_t res = filter_biquad_sat_s32(&filter, 100);
+    TEST_ASSERT_EQUAL(50, res);
+
+    // x[n] = 50
+    // y[0][n] = 1.0 * (  50 + 100 +   0 + 100 +   0  ) = 250
+    // y[1][n] = 1.0 * ( 250 + 100 +   0 + 100 +   0  ) = 450
+    // y[2][n] = 0.5 * ( 450 + 100 +   0 +  50 +   0  ) = 300
+    res = filter_biquad_sat_s32(&filter, 50);
+    TEST_ASSERT_EQUAL(300, res);
+
+    // x[n] = 4000
+    // y[0][n] = 1.0 * ( 4000 +   50 +  100 +  250 +  100  ) = 4500
+    // y[1][n] = 1.0 * ( 4500 +  250 +  100 +  450 +  100  ) = 5400
+    // y[2][n] = 0.5 * ( 5400 +  450 +  100 +  300 +   50  ) = 3150
+    res = filter_biquad_sat_s32(&filter, 4000);
+    TEST_ASSERT_EQUAL(3150, res);
+}
+
+// 8 random biquad filters generated via a python script with 178 samples
```

Comment:
These aren't random or from a python script in lib_xcore_math anymore, we should decide where to document them